wire-ios-sync-engine/Source/UserSession/UserSession.swift

```diff
@@ -0,0 +1,505 @@
+
+    var requiresScreenCurtain: Bool { get }
+
+    /// Whether the app lock on.
+
+    var isAppLockActive: Bool { get set }
+
+    /// Whether the app lock feature is availble to the user.
+
+    var isAppLockAvailable: Bool { get }
+
+    /// Whether the app lock is mandatorily active.
+
+    var isAppLockForced: Bool { get }
+
+    /// The maximum number of seconds allowed in the background before the
+    /// authentication is required.
+
+    var appLockTimeout: UInt { get }
+
+    /// Whether a custom passcode has been set.
+
+    var isCustomAppLockPasscodeSet: Bool { get }
+
+    /// Whether a custom passcode (rather a device passcode) should be used.
+
+    var requireCustomAppLockPasscode: Bool { get }
+
+    /// Whether the user should be notified of the app lock being disabled.
+
+    var shouldNotifyUserOfDisabledAppLock: Bool { get }
+
+    /// Whether the user needs to be informed about configuration changes.
+
+    var needsToNotifyUserOfAppLockConfiguration: Bool { get set }
+
+    /// Unlock the database using the given authentication context.
+
+    func unlockDatabase(with context: LAContext) throws
+
+    /// Open the app lock.
+
+    func openAppLock() throws
+
+    /// Authenticate with device owner credentials (biometrics or passcode).
+    ///
+    /// - Parameters:
+    ///     - passcodePreference: Used to determine which type of passcode is used.
+    ///     - description: The message to dispaly in the authentication UI.
+    ///     - context: The context in which authentication happens.
+    ///     - callback: Invoked with the authentication result.
+
+    func evaluateAppLockAuthentication(
+        passcodePreference: AppLockPasscodePreference,
+        description: String,
+        callback: @escaping (
+            AppLockAuthenticationResult,
+            LAContextProtocol
+        ) -> Void
+    )
+
+    /// Authenticate with a custom passcode.
+    ///
+    /// - Parameter customPasscode: The user inputted passcode.
+    /// - Returns: The authentication result, which should be either `granted` or `denied`.
+
+    func evaluateAuthentication(customPasscode: String) -> AppLockAuthenticationResult
+
+    /// Delete the app lock passcode if it exists.
+
+    func deleteAppLockPasscode() throws
+
+    var conversationDirectory: ConversationDirectoryType { get }
+
+    /// The user who is logged into this session.
+    ///
+    /// This can only be used on the main thread.
+
+    var selfUser: UserType { get }
+
+    var selfLegalHoldSubject: UserType & SelfLegalHoldSubject { get }
+
+    func perform(_ changes: @escaping () -> Void)
+
+    func enqueue(_ changes: @escaping () -> Void)
+
+    func enqueue(
+        _ changes: @escaping () -> Void,
+        completionHandler: (() -> Void)?
+    )
+
+    // TODO: rename to "shouldHideNotificationContent"
+    var isNotificationContentHidden: Bool { get set }
+
+    // TODO: rename to "isEncryptionAtRestEnabled"
```

Comment:
Should we add a ticket, a date or some other reference for when these `TODOs` should be solved?

Reply:
I will take care of it today, while I create the tickets for the rest of the work related to the userSession.